test/unittest/statistics/rtps/RTPSStatisticsTests.cpp

```diff
@@ -0,0 +1,225 @@
+#include <fastdds/rtps/attributes/ReaderAttributes.h>
+#include <fastdds/rtps/attributes/WriterAttributes.h>
+#include <fastdds/rtps/history/WriterHistory.h>
+#include <fastdds/rtps/history/ReaderHistory.h>
+#include <fastdds/rtps/participant/RTPSParticipant.h>
+#include <fastdds/rtps/reader/RTPSReader.h>
+#include <fastdds/rtps/writer/RTPSWriter.h>
+
+#include <statistics/types/types.h>
+
+namespace eprosima {
+namespace fastrtps {
+namespace rtps {
+namespace statistics {
+
+struct MockListener : fastdds::statistics::IListener
+{
+    MOCK_METHOD(void, on_statistics_data, (const fastdds::statistics::Data& ), (override));
+};
+
+/*
+ * This test checks RTPSParticipant, RTPSWriter and RTPSReader statistics module related APIs.
+ *  1. Creates dummy listener objects and associates them to RTPS entities of each kind covering
+ *     the different possible cases: already registered, non-registered, already unregistered.
+ *  2. Verify the different kinds of rtps layer callbacks are performed as expected.
+ */
+TEST(RTPSStatisticsTests, statistics_rpts_listener_management)
+{
+    using namespace std;
+
+    logError(RTPS_STATISTICS, "Test fails because statistics api implementation is missing.");
+
+    // create the entities
+    uint32_t domain_id = 0;
+    RTPSParticipantAttributes p_attr;
+    RTPSParticipant* participant = RTPSDomain::createParticipant(
+        domain_id, true, p_attr);
+    ASSERT_NE(participant, nullptr);
+
+    HistoryAttributes h_attr;
+    h_attr.payloadMaxSize = 255;
+    unique_ptr<WriterHistory> w_history(new WriterHistory(h_attr));
+    unique_ptr<ReaderHistory> r_history(new ReaderHistory(h_attr));
+
+    WriterAttributes w_attr;
+    RTPSWriter* writer = RTPSDomain::createRTPSWriter(participant, w_attr, w_history.get());
+    ASSERT_NE(nullptr, writer);
+
+    ReaderAttributes r_att;
+    RTPSReader* reader = RTPSDomain::createRTPSReader(participant, r_att, r_history.get());
+    ASSERT_NE(nullptr, reader);
+
+    // Check API add and remove interfaces
+    {
+        auto listener1 = make_shared<MockListener>();
+        auto listener2 = make_shared<MockListener>();
+        auto nolistener = listener1;
+        nolistener.reset();
+
+        fastdds::statistics::EventKind kind =
+                fastdds::statistics::EventKind::PUBLICATION_THROUGHPUT;
+        fastdds::statistics::EventKind another_kind =
+                fastdds::statistics::EventKind::SUBSCRIPTION_THROUGHPUT;
+        fastdds::statistics::EventKind yet_another_kind =
+                fastdds::statistics::EventKind::NETWORK_LATENCY;
+
+        // test the participant apis
+        // + fails if no listener has been yet added
+        EXPECT_FALSE(participant->remove_statistics_listener(listener1, kind));
+        // + fails to add an empty listener
+        EXPECT_FALSE(participant->add_statistics_listener(nolistener, kind));
+        // + succeeds to add a new listener
+        ASSERT_TRUE(participant->add_statistics_listener(listener1, kind));
```

Comment:
```suggestion
        EXPECT_TRUE(participant->add_statistics_listener(listener1, kind));
```

Reply:
This one must be an assertion because the following expected results rely on that to be true.